dojo/notifications/views.py

```diff
@@ -129,3 +132,295 @@ def get_scope(self):
     def set_breadcrumbs(self, request: HttpRequest):
         add_breadcrumb(title=_("Template notification settings"), top_level=False, request=request)
         return request
+
+
+class NotificationWebhooksView(View):
+
+    def check_webhooks_enabled(self):
+        if not get_system_setting("enable_webhooks_notifications"):
+            raise Http404
+
+    def check_user_permissions(self, request: HttpRequest):
+        if not request.user.is_superuser:
+            raise PermissionDenied
+        # TODO: finished access for other users
+        # if not user_has_configuration_permission(request.user, self.permission):
+        #     raise PermissionDenied()
+
+    def set_breadcrumbs(self, request: HttpRequest):
+        add_breadcrumb(title=self.breadcrumb, top_level=False, request=request)
+        return request
+
+    def get_form(
+        self,
+        request: HttpRequest,
+        **kwargs: dict,
+    ) -> NotificationsWebhookForm:
+        if request.method == "POST":
+            return NotificationsWebhookForm(request.POST, **kwargs)
+        else:
+            return NotificationsWebhookForm(**kwargs)
+
+    def get(self, request: HttpRequest):
```

Comment:
```suggestion
    def preprocess_request(self, request: HttpRequest):
```
I notice that `post` func in child classes are calling `super().get(request)` and this may be a little unintuitive as it implies a `post` is handling the `get` at the same time. Consider changing the name of this function to something that is more descriptive of the behavior. I suggested `preprocess_request`, but am not wedded to it

Reply:
Done: https://github.com/DefectDojo/django-DefectDojo/pull/7311/commits/56a28e34649135dde2cb0dd6ce24644975b206a4 + https://github.com/DefectDojo/django-DefectDojo/pull/7311/commits/ba36db5af75838b0215ebf673341969f86bae80c